lib/reversetunnel/agent.go

```diff
@@ -165,6 +171,13 @@ type Agent struct {
 	principals []string
 }
 
+// ProxyDetails contains catchable details about the remote proxy.
+type ProxyDetails struct {
```

Comment:
Instead of introducing another struct, can we reuse `PingResponse` received from the ping call? Or at least `ProxySettings` part of it.

Reply:
In situation when `Addr` is legacy reverse tunnel port and ping call failed we would need to create fake ProxySettings struct and set `ALPNSNIListenerEnabled` = false thought  `ProxySettings` is name is a bit misleading and I have renamed it to ReverserTunnelDetails